firewood/src/merkle.rs

```diff
@@ -1388,6 +1672,39 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn iterate_empty() {
+        let merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+        let mut it = merkle.get_iter(Some(b"x"), root).unwrap();
+        let next = it.next().await;
+        assert!(next.is_none())
+    }
+
+    #[tokio::test]
+    async fn iterate_many() {
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        // insert all values from u8::MIN to u8::MAX, with the key and value the same
+        for k in u8::MIN..=u8::MAX {
+            merkle.insert([k], vec![k], root).unwrap();
+        }
+
+        for start in [Some([u8::MIN]), None] {
+            let mut it = merkle.get_iter(start, root).unwrap();
```

Comment:
It would be better to have separate unit tests for a variety of start cases. 
 - no starting key
 - a starting key lower than the first key in the trie
 - a starting key at the first key in the trie
 - a starting key greater than the starting key in the trie

Reply:
I didn't say that Case 1 wasn't covered, I said 
>It would be better to have separate unit tests

You can use `test_case` to have separate inputs and thus separate tests instead of looping over a two-element array. 

>I'm not sure why we need case 4.

To make sure that the code behaves as expected, especially in the event of a change. We need to make sure that keys are skipped properly.